src/components/Button.svelte

```diff
@@ -0,0 +1,185 @@
+<script lang="ts">
+  export let id: string | null = null;
+  export let testId: string | null = null;
+  export let variant: 'primary' | 'secondary' | 'text' | 'danger' | 'link' =
+    'primary';
+  export let size: 'small' | 'medium' | 'large' = 'medium';
+  export let block = false;
+  export let style = ``;
+  export let loading = false;
+  export let disabled = false;
+</script>
+
+<button
+  disabled={disabled || loading}
+  data-test-id={testId}
+  {id}
+  {style}
+  class:block={!!block}
+  class:large={size === 'large'}
+  class:small={size === 'small'}
+  class:primary={variant === 'primary'}
+  class:secondary={variant === 'secondary'}
+  class:danger={variant === 'danger'}
+  class:text={variant === 'text'}
+  class:link={variant === 'link'}
+  on:click
```

Comment:
holy god button batman

Reply:
(not a negative comment, its fine, ship it)

Reply:
Nah, I'll refactor all this stuff and request a code review at the end. Just wanted to get the go-ahead that this PR was actually desired before spending time refining this stuff.